ibm/resource_ibm_iam_user_policy.go

```diff
@@ -0,0 +1,376 @@
+							Optional: true,
+						},
+						"space_guid": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"organization_guid": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"roles": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMIAMUserPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	iamClient, err := meta.(ClientSession).IAMAPI()
+	if err != nil {
+		return err
+	}
+	accClient, err := meta.(ClientSession).BluemixAcccountAPI()
+	if err != nil {
+		return err
+	}
+	accountGuid := d.Get("account_guid").(string)
+
+	userEmail := d.Get("ibm_id").(string)
+	account, err := accClient.Accounts().Get(accountGuid)
+	if err != nil {
+		return fmt.Errorf("Error retrieving account: %s", err)
+	}
+	userId, err := getIBMUniqueIdOfUser(accountGuid, userEmail, meta)
+	if userId == "" || err != nil {
+		return fmt.Errorf("User %q does not exist in the account:%q", userEmail, account.Name)
+	}
+
+	roleIdSet := d.Get("roles").(*schema.Set)
+	roles := getRoles(roleIdSet)
+
+	policyServices := d.Get("resources").(*schema.Set)
+	resources := createResources(policyServices, iamClient, accountGuid)
+
+	params := v1.AccessPolicyRequest{
+		Roles:     roles,
+		Resources: resources,
+	}
+
+	accessPolicyResponse, etag, err := iamClient.IAMPolicy().Create(accountGuid, userId, params)
+	if err != nil {
+		return err
+	}
+	d.SetId(accessPolicyResponse.ID)
+	d.Set("account_guid", accountGuid)
+	d.Set("ibm_id", userEmail)
+	d.Set("etag", etag)
+
+	return resourceIBMIAMUserPolicyRead(d, meta)
+}
+
+func resourceIBMIAMUserPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	iamClient, err := meta.(ClientSession).IAMAPI()
+	if err != nil {
+		return err
+	}
+	accClient, err := meta.(ClientSession).BluemixAcccountAPI()
+	if err != nil {
+		return err
+	}
+	rolesMaps := make(map[string]string)
+	rolesMaps[VIEWER_ID] = VIEWER
+	rolesMaps[ADMINISTRATOR_ID] = ADMINISTRATOR
+	rolesMaps[EDITOR_ID] = EDITOR
+	rolesMaps[OPERATOR_ID] = OPERATOR
+
+	accountGuid := d.Get("account_guid").(string)
+
+	userEmail := d.Get("ibm_id").(string)
+	account, err := accClient.Accounts().Get(accountGuid)
+	if err != nil {
+		return fmt.Errorf("Error retrieving account: %s", err)
+	}
+	userId, err := getIBMUniqueIdOfUser(accountGuid, userEmail, meta)
+	if userId == "" || err != nil {
+		return fmt.Errorf("User %q does not exist in the account:%q", userEmail, account.Name)
+	}
+	policyId := d.Id()
+	iamPolicy, err := iamClient.IAMPolicy().Get(accountGuid, userId, policyId)
+	if err != nil {
+		return fmt.Errorf("Unable to read policy:%s", err)
+	}
+
+	resources := iamPolicy.Resources
+	roles := iamPolicy.Roles
+
+	d.Set("roles", flattenIAMPolicyRoles(roles, rolesMaps))
```

Comment:
I think we don't need to pass rolesMaps in the function. Have that map defined somewhere in var